### PR TITLE
kak-lsp: 8.0.0 -> 9.0.0

### PR DESCRIPTION
**Pull request by wahjava**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/kakoune/plugins/kak-lsp.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  cargoSha256 = "0g67s6n45rxvv1q5s7x5ajh5n16p68bhlsrsjp46qamrraz63d68";
+
+  meta = with lib; {
+    description = "Kakoune Language Server Protocol Client";
+    homepage = "https://github.com/kak-lsp/kak-lsp";
+    license = licenses.unlicense;
+    maintainers = with maintainers; [ abbe ];
+    platform = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/kakoune/plugins/kak-lsp.nix:21]:
```suggestion
```

set by buildRustPackage

Comment (wahjava) [reply pkgs/applications/editors/kakoune/plugins/kak-lsp.nix]: Updated, thanks!

**Comment by wahjava** [comment]
Thanks for pointing to PR #108363. It seems [kak-lsp is already part of nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L5397), just not part of `kakounePlugins`, where I was expecting it to be present. I will submit a new diff to update existing derivation instead. Should I open a new PR, or overwrite this one ?

**Comment by SuperSandro2000** [comment]
> Should I open a new PR, or overwrite this one ?

Overwriting and changing the title would be fine for me.

**Comment by wahjava** [comment]
> > Should I open a new PR, or overwrite this one ?
> 
> Overwriting and changing the title would be fine for me.

Pushed, thanks!

**Comment by Flakebi** [comment]
Can you reuse the top-level kak-lsp package like it is done for parinfer-rust and rep?

**Comment by wahjava** [comment]
> Can you reuse the top-level kak-lsp package like it is done for parinfer-rust and rep?

Sorry, I'm confused. This PR (the latest commit) changes the file `pkgs/tools/misc/kak-lsp/default.nix` and that file is already referenced in `pkgs/top-level/all-packages.nix`. If this is not what you expect, then could you please elaborate ?

Thanks!

**Comment by Flakebi** [comment]
The change to `pkgs/tools/misc/kak-lsp/default.nix` looks good, thanks for the update.

You add the same file again at `pkgs/applications/editors/kakoune/plugins/kak-lsp.nix`, which is not necessary.
You can add `kak-lsp` to `kakounePlugins` with `inherit kak-lsp;` in the beginning of `pkgs/applications/editors/kakoune/plugins/default.nix` (just add it to the list of `inherit parinfer-rust rep;` and to the package inputs in the first line).
`inherit pkgname;` is equivalent to `pkgname = pkgname;`: It adds an attribute with the same name as a variable from the current scope.

**Comment by wahjava** [comment]
> The change to `pkgs/tools/misc/kak-lsp/default.nix` looks good, thanks for the update.
> 
> You add the same file again at `pkgs/applications/editors/kakoune/plugins/kak-lsp.nix`, which is not necessary.
> You can add `kak-lsp` to `kakounePlugins` with `inherit kak-lsp;` in the beginning of `pkgs/applications/editors/kakoune/plugins/default.nix` (just add it to the list of `inherit parinfer-rust rep;` and to the package inputs in the first line).
> `inherit pkgname;` is equivalent to `pkgname = pkgname;`: It adds an attribute with the same name as a variable from the current scope.

**Comment by Flakebi** [comment]
You need to add kak-lsp to the arguments, otherwise it will not be found.

```
error: --- UndefinedVarError ----------------------------------------------------------------------------------------------------------- nix-env
at: (3:2) in file: /home/sebi/.cache/nixpkgs-review/pr-110663/nixpkgs/pkgs/applications/editors/kakoune/plugins/default.nix

     2|
     3| {
      |  ^
     4|   inherit parinfer-rust rep kak-lsp;

undefined variable 'kak-lsp'
```

**Comment by wahjava** [comment]
> You need to add kak-lsp to the arguments, otherwise it will not be found.
> 
> ```
> error: --- UndefinedVarError ----------------------------------------------------------------------------------------------------------- nix-env
> at: (3:2) in file: /home/sebi/.cache/nixpkgs-review/pr-110663/nixpkgs/pkgs/applications/editors/kakoune/plugins/default.nix
> 
>      2|
>      3| {
>       |  ^
>      4|   inherit parinfer-rust rep kak-lsp;
> 
> undefined variable 'kak-lsp'
> ```

Ofcourse, and sorry, I overlooked that. It now builds fine when referenced with `kakounePlugins.kak-lsp`.

Thanks!

**Review by Flakebi — APPROVED** [review]
Thank you!

Result of `nixpkgs-review pr 110663` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kak-lsp</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110663` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>kak-lsp</li>
</ul>
</details>

File: pkgs/tools/misc/kak-lsp/default.nix [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
     repo = pname;
     rev = "v${version}";
-    sha256 = "0nka51szivwhlfkimjiyzj67nxh75m784c28ass6ihlfax631w9m";
+    sha256 = "1wfv2fy5ga6kc51zka3pak0hq97csm2l11bz74w3n1hrf5q9nnf8";
   };
 
-  cargoSha256 = "174qy50m9487vv151vm8q6sby79dq3gbqjbz6h4326jwsc9wwi8c";
+  cargoSha256 = "0g67s6n45rxvv1q5s7x5ajh5n16p68bhlsrsjp46qamrraz63d68";
 
   buildInputs = lib.optional stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/kak-lsp/default.nix:16]:
```suggestion
  buildInputs = lib.optional stdenv.isDarwin [ Security ];
```

Please add Security to inputs and inherit it from top level ``inherit (darwin.apple_sdk.frameworks) Security``.

Comment (wahjava) [reply pkgs/tools/misc/kak-lsp/default.nix]: Updated in 35a2887. I'm only able to test this on NixOS, but let me know if this is not how you expected it.

Comment (SuperSandro2000) [reply pkgs/tools/misc/kak-lsp/default.nix]: I meant like this https://github.com/NixOS/nixpkgs/commit/a72353735c1ca6e157e257318f3cda4649ea645e

Comment (wahjava) [reply pkgs/tools/misc/kak-lsp/default.nix]: Thanks, updated in 25e2b72. Let me know if you like me to note this change in commit message as well, although I don't know what this change will be described so as to be useful in a commit message.

**Comment by SuperSandro2000** [comment]
@wahjava please fix the merge conflitct.

**Comment by wahjava** [comment]
> @wahjava please fix the merge conflitct.

Merged, and forced pushed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110663` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>kak-lsp</li>
</ul>
</details>

**Pull request merged**